code/game/objects/items/weapons/grenades/flashbang.dm

```diff
@@ -5,120 +5,117 @@
-		if (locate(/obj/item/weapon/cloaking_device, M))			// Called during the loop that bangs people in lockers/containers and when banging
-			for(var/obj/item/weapon/cloaking_device/S in M)			// people in normal view.  Could theroetically be called during other explosions.
-				S.active = 0										// -- Polymorph
-				S.icon_state = "shield0"
+/obj/item/weapon/grenade/flashbang/proc/bang(var/turf/T , var/mob/living/carbon/M)  // Added a new proc called 'bang' that takes a location and a person to be banged.
+	if (locate(/obj/item/weapon/cloaking_device, M))								// Called during the loop that bangs people in lockers/containers and when banging
+		for(var/obj/item/weapon/cloaking_device/S in M)								// people in normal view.  Could theroetically be called during other explosions.
+			S.active = 0															// -- Polymorph
+			S.icon_state = "shield0"
 
-		M << "<span class='danger'>BANG</span>"
-		playsound(src.loc, 'sound/weapons/flashbang.ogg', 50, 1, 5)
+	M << "<span class='danger'>BANG</span>"
+	playsound(src.loc, 'sound/weapons/flashbang.ogg', 50, 1, 5)
 
 //Checking for protections
-		var/eye_safety = 0
-		var/ear_safety = 0
-		if(iscarbon(M))
-			eye_safety = M.eyecheck(TRUE)
-			if(ishuman(M))
-				if(istype(M:l_ear, /obj/item/clothing/ears/earmuffs) || istype(M:r_ear, /obj/item/clothing/ears/earmuffs))
-					ear_safety += 2
-				if(HULK in M.mutations)
-					ear_safety += 1
-				if(istype(M:head, /obj/item/clothing/head/helmet))
-					ear_safety += 1
+	var/eye_safety = 0
+	var/ear_safety = 0
+	if(iscarbon(M))
+		eye_safety = M.eyecheck(TRUE)
+		if(ishuman(M))
+			if(istype(M:l_ear, /obj/item/clothing/ears/earmuffs) || istype(M:r_ear, /obj/item/clothing/ears/earmuffs))
+				ear_safety += 2
+			if(HULK in M.mutations)
+				ear_safety += 1
+			if(istype(M:head, /obj/item/clothing/head/helmet))
+				ear_safety += 1
 
 //Flashing everyone
-		if(eye_safety < FLASH_PROTECTION_MODERATE)
-			flick("e_flash", M.flash)
-			M.Stun(2)
-			M.Weaken(10)
-				//Vaurca damage 15/01/16
-			var/mob/living/carbon/human/H = M
-			if(isvaurca(H))
-				var/obj/item/organ/eyes/E = H.get_eyes()
-				if(!E)
-					return
-				usr << span("alert", "Your eyes burn with the intense light of the flash!.")
-				E.damage += rand(10, 11)
-				if(E.damage > 12)
-					M.eye_blurry += rand(3,6)
-				if (E.damage >= E.min_broken_damage)
-					M.sdisabilities |= BLIND
-				else if (E.damage >= E.min_bruised_damage)
-					M.eye_blind = 5
-					M.eye_blurry = 5
-					M.disabilities |= NEARSIGHTED
-					spawn(100)
-						M.disabilities &= ~NEARSIGHTED
-
-
+	if(eye_safety < FLASH_PROTECTION_MODERATE)
+		flick("e_flash", M.flash)
+		M.Stun(2)
+		M.Weaken(10)
+			//Vaurca damage 15/01/16
+		var/mob/living/carbon/human/H = M
+		if(isvaurca(H))
+			var/obj/item/organ/eyes/E = H.get_eyes()
+			if(!E)
+				return
+			usr << span("alert", "Your eyes burn with the intense light of the flash!.")

```

Comment:
excuse, `usr`, here? Who is supposed to get this message, `H` I presume. Replace `usr` with `H`.